### PR TITLE
Fix watcher conditional logic

### PR DESCRIPTION
**Pull request by erikerlandson**
I ran into some exceptions being thrown from the endpoint watcher in the submission client.  It turned out that `null` values were being returned for `endpoints.getSubsets`, instead of just empty collection.  I added some guarding checks for `null`, which fixed the problem.

I also modified the parentheses around action types, because that appeared incorrect.  I believe it should be checking for `(action == added || action == modified)` as a unit in the first clause.

### CODE REVIEW

**Comment by erikerlandson** [comment]
Anybody else getting build fail?  It's failing in core, which this PR makes no changes to.

**Comment by erikerlandson** [comment]
jenkins please test this

**Comment by foxish** [comment]
rerun unit tests please

**Comment by foxish** [comment]
@erikerlandson I think it's a flake. The rerun should fix it.

**Comment by erikerlandson** [comment]
Do these edits look OK to everyone else?

**Comment by foxish** [comment]
Taking a look now. Sorry about the delay.

**Comment by foxish** [comment]
I think `endpoints` itself can't be null, but defensively, it seems like a good idea to check for it. Agree on the parens being incorrect before. LGTM

File: resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/Client.scala [hunk 1]
```diff
@@ -554,7 +556,7 @@ private[spark] class Client(
   private class DriverServiceReadyWatcher(resolvedDriverService: SettableFuture[Service])
       extends Watcher[Service] {
     override def eventReceived(action: Action, service: Service): Unit = {
-      if ((action == Action.ADDED) || (action == Action.MODIFIED)
+      if ((action == Action.ADDED || action == Action.MODIFIED)
```
Comment (ash211) [inline resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/v1/Client.scala:559]: the parentheses change slightly modifies the evaluation order of these expressions, and the new ordering seems correct.

Java precedence for boolean expressions is `!` then `&&` then `||` so previously this was evaluating the `&&` first and then the `||`

**Review by ash211 — APPROVED** [review]
LGTM let's merge when the build passes

**Pull request merged**